graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/requests/IndexSetUpdateRequest.java

```diff
@@ -98,26 +109,31 @@ public static IndexSetUpdateRequest fromIndexSetConfig(IndexSetConfig indexSet)
                 indexSet.rotationStrategyClass(),
                 indexSet.rotationStrategy(),
                 indexSet.retentionStrategyClass(),
-                indexSet.retentionStrategy());
+                indexSet.retentionStrategy(),
+                indexSet.indexOptimizationMaxNumSegments(),
+                indexSet.indexOptimizationDisabled());
 
     }
 
     public IndexSetConfig toIndexSetConfig(IndexSetConfig oldConfig) {
-        return IndexSetConfig.builder()
-                .id(id())
-                .title(title())
-                .description(description())
-                .isWritable(isWritable())
-                .indexPrefix(oldConfig.indexPrefix())
-                .indexMatchPattern(oldConfig.indexMatchPattern())
-                .indexWildcard(oldConfig.indexWildcard())
-                .shards(shards())
-                .replicas(replicas())
-                .rotationStrategyClass(rotationStrategyClass())
-                .rotationStrategy(rotationStrategy())
-                .retentionStrategyClass(retentionStrategyClass())
-                .retentionStrategy(retentionStrategy())
-                .creationDate(oldConfig.creationDate())
-                .build();
+        return IndexSetConfig.create(
```

Comment:
At one point we should agree on how we do auto value object creation so we don't change it in every PR. :wink: